vignettes/arx-classifier-vignette.Rmd

```diff
@@ -0,0 +1,161 @@
+---
+title: "arx_classifier"
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  echo = TRUE,
+  collapse = FALSE,
+  comment = "#>",
+  warning = FALSE,
+  message = FALSE,
+  cache = TRUE
+)
+```
+
+## Load required packages
+
+```{r, message = FALSE, warning = FALSE}
+library(dplyr)
+library(purrr)
+library(ggplot2)
+library(epipredict)
+```
+
+## Introducing the ARX classifier
+
+The `arx_classifier()` function is an autoregressive classification model for `epi_df` data that is used to predict a discrete class for each case under consideration. It is a direct forecaster in that it estimates the classes at a specific horizon or ahead value.
+
+To get a sense of how the `arx_classifier()` works, let's consider a simple example with minimal inputs. For this, we will use the built-in `case_death_rate_subset` that contains confirmed COVID-19 cases and deaths from JHU CSSE for all states over Dec 31, 2020 to Dec 31, 2021. From this, we'll take a subset of data for five states over June 4, 2021 to December 31, 2021. Our objective is to predict whether the case rates are increasing when considering the 0, 7 and 14 day case rates:
+
+```{r}
+jhu <- case_death_rate_subset %>%
+  dplyr::filter(
+    time_value >= "2021-06-04",
+    time_value <= "2021-12-31",
+    geo_value %in% c("ca", "fl", "tx", "ny", "nj")
+  )
+
+out <- arx_classifier(jhu,
+  outcome = "case_rate",
+  predictors = "case_rate"
```

Comment:
We could... But since this vignette is supposed to be an adaption of the classification problem in the preprocessing and models vignette, I'm using the same set-up in terms of data, outcome, and predictors, that is used there. Part of the motivation is to show that we can reduce that manual implementation of the classifier for that problem from about 30 down to 3 lines of code by using `arx_classifier()`